packages/cosmic-swingset/lib/daemon/main.go

```diff
@@ -116,7 +116,8 @@ func makeNewApp(sendToController Sender) func(logger log.Logger, db dbm.DB, trac
 			baseapp.SetHaltTime(viper.GetUint64(server.FlagHaltTime)),
 			baseapp.SetInterBlockCache(cache),
 		)
-		msg := `{"type":"AG_COSMOS_INIT"}`
+		msg := fmt.Sprintf(`{"type":"AG_COSMOS_INIT","ibcPort":%d}`, abci.IBCPort)
```

Comment:
Assuming this is a temporary expedient and performance nor form are concerns. This should work in general.